Toshi/Source/Toshi/File/TTRB.cpp

```diff
@@ -64,45 +56,40 @@ namespace Toshi
 
 				if (sectionName == TMAKEFOUR("HEAD"))
 				{
-					TTODO("HEAD section");
-
 					int numsections = (sectionSize - 4) / 0xC;
 					m_pHeader = static_cast<Header*>(tmalloc(sizeof(Header) + sizeof(SecInfo) * numsections));
 					m_pHeader->m_ui32Version = 0;
 
 					ttsf.ReadBytes(&m_pHeader->m_i32SectionCount, sizeof(m_pHeader->m_i32SectionCount));
 
-					// TASSERT(m_pHeader->m_ui32Version == TMAKEVERSION(0, 0), "HEAD section cannot have a version");
 					TASSERT(m_pHeader->m_i32SectionCount == numsections, "HEAD section has wrong num of sections");
 
-					SecInfo* pCur = reinterpret_cast<SecInfo*>(m_pHeader + 1);
-					for (size_t i = 0; i < m_pHeader->m_i32SectionCount; i++)
+					SecInfo* pSect = &m_pHeader->operator[](0);
```

Comment:
&m_pHeader[0]?

Reply:
operator[] doesn't work with pointers

Reply:
Hm then i would call it GetSection(int index) or smth like that. In my opinion that would look better